src/Type-Transformation/resolutions/05/1-pass-func-overload.ts

```diff
@@ -0,0 +1,53 @@
+import { Equal, Expect } from "../../../../helper";
+import { expect, it } from "vitest";
+
+/*
+  NOTE: one thing to note, is that the implemantation body (the signature of the function)
+  isn't exposed outside of the function, so it's parameters becomes only an internal signature 
+*/
+
+
+// 1. same as before, if we pass to the function goodbye it must return hello and vice versa,
+// we want a function overload that can handle the diferent cases
+
+
+type TGreetingOutput = "hello" | "goodbye";
+
+type TGreetingInput = TGreetingOutput extends infer T
+    ? T extends "hello"
+        ? "goodbye"
+        : T extends "goodbye"
+        ? "hello"
+        : never
+    : never;
+
```

Comment:
este tipo esta de mas, porque ya estas discriminando desde el tipo, la idea de las funciones overloads es asignarle la logica a las funciones.
basta con asignarles una union 
```function youSayGoodbyeISayHello(greeting: 'hello' | 'goobye')```

Reply:
Custom types remove, fix done in next commit.